prow/resultstore/uploader.go

```diff
@@ -35,13 +35,19 @@ func NewUploader(client *Client) *Uploader {
 
 // Upload uploads a completed Prow job's results to ResultStore's API:
 // https://github.com/googleapis/googleapis/blob/master/google/devtools/resultstore/v2/resultstore_upload.proto
+// If error is returned, the call should be retried later.
```

Comment:
Perhaps "This function distinguishes between transient and permanent errors coming from ResultStore, and will only return error for what it thinks are transient errors, so that callers can safely retry without fear of getting stuck" is more to the point.